src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/ExecutableStatementCountCheck.java

```diff
@@ -95,7 +95,6 @@ public void setMax(int max) {
     @Override
     public void beginTree(DetailAST rootAST) {
         context = new Context(null);
-        contextStack.clear();
```

Comment:
We should rely that visit a leave methods are called.
But there is nuance , if any exception happen in logic of Check, check will accumulate incorrect state.
In Checker we have option `haltOnException` that might not fail execution on first exception. So next file will be processed, and all Checks need to reset their state on beginTree. In other case there might be extra false positives and complete mess in validation after exception.

This situation is impossible to reproduce, such code is good to have.
Let's make a method `cleanupCheckInternalState` and such non required cleanup in it, additionally configure pitest to skip such method at all.